frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/storage-efficiency-card/storage-efficiency-card-item.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import { FieldLevelHelp, humanizeBinaryBytes } from '@console/internal/components/utils';
+
+const ItemBody: React.FC<ItemBodyProps> = React.memo(
+  ({ title, stats, infoText, isLoading, error }) => {
+    let status: React.ReactElement;
+    if (isLoading) {
+      status = <div className="skeleton-text ceph-storage-efficiency-card__item-body--loading" />;
+    } else if (error || !stats) {
+      status = <span className="co-dashboard-text--small text-muted">Not available</span>;
+    } else {
+      status = <span className="ceph-storage-efficiency-card__item-text">{stats}</span>;
+    }
+    return (
+      <div className="co-inventory-card__item">
+        <div className="ceph-storage-efficiency-card__item-title">{title}</div>
+        <div className="ceph-storage-efficiency-card__item-status">
+          {status}
+          <FieldLevelHelp>{infoText}</FieldLevelHelp>
+        </div>
+      </div>
+    );
+  },
+);
```

Comment:
We can make this component more generic, renaming to `StorageEfficiencyItem` and avoid using `SavedItem`, `CompressionItem`.

You can pass one `handler` function to it to get the stats:

```
({ title, data, infoText, isLoading, error, handler }) => {
    let status: React.ReactElement;
    const stats = handler(data);
```
The handler will be your extracted logic for :
```
   if (ratio) {
      const capacityRatio = Number(ratio);
      stats = capacityRatio === 0 ? '1:1' : `${Math.round(capacityRatio)}:1`;
    }

  if (ratio) {
      const capacityRatio = Number(ratio);
      stats = capacityRatio === 0 ? '1:1' : `${Math.round(capacityRatio)}:1`;
    }

```
which you can pass as prop;

You can call directly `StorageEfficiencyItem` as:
```
<DashboardCardBody className="co-dashboard-card__body--no-padding">
        <StorageEfficiencyItem {...compressionRatioProps} />
        <StorageEfficiencyItem {...savingsProps} />
    </DashboardCardBody>
```